From e521477a387a9ba2e22ff735dc2b84a86a22cc10 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Sat, 9 Dec 2023 16:48:27 +0200
Subject: [PATCH] tests: Wait a bit after radio work start in dpp_listen()

Even though the radio work has started here, kernel might not have
started ROC (i.e., the REMAIN_ON_CHANNEL event has not yet been
received). Wait a bit to increase likelihood of everything being ready
for the following DPP operation.

Signed-off-by: Jouni Malinen <j@w1.fi>
---
 tests/hwsim/wpasupplicant.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/hwsim/wpasupplicant.py b/tests/hwsim/wpasupplicant.py
index b9764baff..8e8664299 100644
--- a/tests/hwsim/wpasupplicant.py
+++ b/tests/hwsim/wpasupplicant.py
@@ -1556,6 +1556,7 @@ class WpaSupplicant:
         work_started = "dpp-listen@" + self.ifname + ":" + str(freq) + ":1"
         for i in range(10):
             if work_started in self.request("RADIO_WORK show"):
+                time.sleep(0.0005)
                 return
             time.sleep(0.01)
         raise Exception("Failed to start DPP listen work")
-- 
2.43.0

